### ISSUE TITLE
Ability to run on headless Linux server -- remove Tk dependency?

### ISSUE THREAD

**Issue by taa1** [okko/perl-Device-BlinkyTape#2]
Hi Okko,

I'm running Fedora 23, 64 bit, on a headless server and want to use a connected BlinkyTape for various status/progress indications. The problem is that installing this perl module requires Tk for the simulator feature, which depends on X Windows, which I don't plan to run because I don't need the simulator. 

I have the example script without `simulate => 1` and with the actual serial device:

~~~perl
my $bb = Device::BlinkyTape::WS2811->new(dev => '/dev/ttyACM0');
~~~

but running it still expects the Tk module so I can't try out the script.

Would it be possible for you to create a version of this perl module that doesn't require Tk?

(FYI, I was the person who reported the problem [Running Device::BlinkyTape on Windows](https://rt.cpan.org/Ticket/Display.html?id=96610) but like you I didn't receive an email notification that you'd responded there. I will plan to test the Windows version since you were kind enough to respond to my problem report.)

**Comment by okko** [comment]
The simulation with Tk should be an optional feature.

-Can you give me steps to reproduce and the error you got?

-Can you try if it works if you comment out this line?
https://github.com/okko/perl-Device-BlinkyTape/blob/master/lib/Device/BlinkyTape.pm#L17

**Comment by taa1** [comment]
Gah, I didn't have the new `BlinkyTape.pm` in place.

Same [blinkytape-test2.pl](http://pastebin.com/SLzmjhyB) as being used in issue #3 

Results from running : http://pastebin.com/W11twgUt

~~~
$ perl -v

This is perl 5, version 24, subversion 1 (v5.24.1) built for x86_64-linux

$perl -e 'print "$^O\n\n;"'
linux
~~~

As far as steps to reproduce, there's not much to it that I can think of:

1. Install Fedora 23, 64 bit, minimal server
2. make sure perl version is 5.24.1
3. run blinkytape-test2.pl and use `cpan` to install missing modules

**Comment by okko** [comment]
Thanks! It should (might) work if you comment out that TK requirement, did you try it?

**Comment by okko** [comment]
You could also try this bleeding edge release now: https://github.com/okko/perl-Device-BlinkyTape/releases/tag/v0.005

**Comment by UNIVAC-Colonel-Panic** [comment]
Yes, I did try commenting out line 17 (see [pastebin](http://pastebin.com/W11twgUt) at line 23 that proves it is commented out), but I couldn't get past the endless loop of the Device::SerialPort dependency. 

I'll try the bleeding edge release next. :D

**Comment by UNIVAC-Colonel-Panic** [comment]
Hi Okko,

I'm having trouble installing. I'm trying:

~~~
cpanm --sudo https://github.com/okko/perl-Device-BlinkyTape/archive/v0.005.tar.gz
~~~

but get the message:

~~~
(snip)
Fetching https://github.com/okko/perl-Device-BlinkyTape/archive/v0.005.tar.gz
-> OK
Unpacking v0.005.tar.gz
Entering perl-Device-BlinkyTape-0.005
Configuring perl-Device-BlinkyTape-0.005
-> N/A
-> FAIL The distribution doesn't have a proper Makefile.PL/Build.PL
~~~

What am I doing wrong?

**Comment by UNIVAC-Colonel-Panic** [comment]
Hi Okko,

Could you give me the proper steps to install the new version from github? I don't know enough about Perl module packages. I did spend several hours searching on Google to try to figure it out on my own. I found and tried `cpanm` but without success as you can see above.

**Comment by UNIVAC-Colonel-Panic** [comment]
Hi Okko,

It's been several weeks so I'm checking in.... Can you please let me know how I should be installing the new version from github?
